lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4574,8 +4576,14 @@ ParseResult FIRCircuitParser::parseExtModule(CircuitOp circuit,
                         : Convention::Internal;
   auto conventionAttr = ConventionAttr::get(getContext(), convention);
   auto annotations = ArrayAttr::get(getContext(), {});
-  builder.create<FExtModuleOp>(info.getLoc(), name, conventionAttr, portList,
-                               defName, annotations, parameters, internalPaths);
+  auto extModuleOp = builder.create<FExtModuleOp>(
+      info.getLoc(), name, conventionAttr, portList, defName, annotations,
+      parameters, internalPaths);
+
+  auto isMainModule = (name == circuit.getName());
+  auto visibility = isMainModule ? SymbolTable::Visibility::Public
```

Comment:
Looks like the main-module-is-extmodule case gets the convention controlled by "scalarize-ext-modules" not the one for top, but this is pre-existing and maybe going away soon anyway.

Otherwise, I would suggest this code also sets convention according to the "scalarize-public-modules" option, when the extmodule is public (or error if those options are inconsistent?).  Anyway.

Reply:
I agree that this should be cleaned up. I'll change it such that it will also get turned on if public modules are scalarized. 

I.e., an external module that is the top module is a public module.